cmd/terminate.go

```diff
@@ -111,20 +114,21 @@ func terminateNonInteractive(h *ec2helper.EC2Helper) {
 		instanceIdFlag[i] = strings.TrimSpace(instanceIdFlag[i])
 	}
 
-	cli.ShowError(h.TerminateInstances(instanceIdFlag), "Terminating instances failed")
+	instFilters, err := tag.GetTagAsFilter(flagConfig.UserTags)
+	instancesToTerm, err := h.GetInstancesByFilter(instanceIdFlag, instFilters)
+	if err != nil {
+		cli.ShowError(err, "Finding instances with filters failed")
+		return
+	}
+
+	cli.ShowError(h.TerminateInstances(instancesToTerm), "Terminating instances failed")
 }
 
 // Validate flags using some simple rules. Return true if the flags are validated, false otherwise
 func ValidateTerminateFlags() bool {
-	if !isInteractive && instanceIdFlag == nil && regionFlag == "" {
-		fmt.Println("Not in interactive mode and no flag is specified")
+	if !isInteractive && instanceIdFlag == nil && len(flagConfig.UserTags) == 0 {
```

Comment:
> Does this change any user facing behaviors of terminate? / Or was region just not necessary when instance Ids are input by user?

Slightly changes behavior-- when users don't include `--region` flag, then terminate will use the default region per their configuration. Although region wasn't necessary it is possible this was done intentionally to prevent mistakenly terminating instances? Any thoughts?

> What happens when the same tag was used on instances in multiple regions? Will all of those instances be terminated?

No, global termination has not been implemented yet. The region provided by user (flag, interactive selection, or aws configure) will be the only region in which instances get terminated.

Once I implement the global operations, then I'm thinking to make `--region=all` a valid option and that would be the only way to terminate instances with same tag in multiple regions

Reply:
Gotcha! Then, it probably makes sense to force the region flag? That way, no change in behavior and it remains explicit.

Reply:
I actually misspoke, region was not required before these changes.

`simple-ec2 terminate -n <instanceId>` will behave the same before and after these changes: delete instanceId in default region